graph_snyk_cve_sync.py

```diff
@@ -313,6 +314,11 @@ def _extract_data_from_feed(self):
             else:
                 logger.info("Ignoring the ecosystem {} from the feed".format(eco))
 
+    def _is_date_in_range(self, date):
+        """Check to see if the date of vuln falls in the range."""
+        date_obj = datetime.strptime(date.split('T')[0], '%Y-%m-%d')
+        return date_obj >= self.start_day
```

Comment:
This will ingest even for today, dont we want to restrict it to yesterday?

Reply:
plan is to schedule the cron at 00 so there wont be anything for today

Reply:
@prashbnair changes done

Reply:
what is the utc check for, that will always be true.
why dont we want todays data to be ingested?

Reply:
this contradicts your prev statement. 
We dont ingest today's vuln because snyk keeps updating it several times a day. We wait for all the updates to happen and then ingest it the next day
So suppose the offset is 2 days, then it will pick all the vulns for date 6th Aug, 5th Aug
7th aug > vulns_modification_time >= 5th Aug

Reply:
If you take the time now, that is always going to be greater than the vuln modification time. 
When I made my earlier statement you did not have that check.

Reply:
date_obj = datetime.strptime(date.split('T')[0], '%Y-%m-%d') -> this will only pick the date and doesnt consider the time

Reply:
maybe im not getting the point you are trying to make. a small example might help

Reply:
@prashbnair i think i got the point you were trying to make. Actually, as i had intended for this to run at midnight, it would not cause the problem you are trying to mention. But yes, running at any other hours might cause a problem. Hence i have added a logic to take only the date when taking the value for utcnow

Reply:
Ok, let me explain with an example.
utc.now at midnight today will be 08/08/2020
Will we have a modification time greater than that, from a file that was sent yesterday?

Reply:
so here is the scenario. Lets say it runs at midnight 08/08/20
so utc_now = 08-08-2020. If we set offset as 3 days, start_day=05-08-2020.
Now when we fetch feed, we get vulns with modified date as follow 01-08-2020, 04-08-2020, 06-08-2020, 08-08-2020
so by this logic
**self.utc_now > date_obj >= self.start_day**
It will pick only **06-08-2020** . The vuln 08-08-2020 will eventually get ingested in the next run i.e 09-08-2020 midnight. In case its still unclear, we can have a short 5 min call on monday and talk about it :)

Reply:
Lets talk about this.